tests/vmi_multus_test.go

```diff
@@ -856,13 +856,14 @@ var _ = Describe("[Serial]SRIOV", func() {
 			//create two vms on the smae sriov network
 			vmi1, vmi2 := createSriovVMs(sriovnet3, sriovnet3, cidrA, cidrB)
 
-			assert.XFail("suspected cloud-init issue: https://github.com/kubevirt/kubevirt/issues/4642")
-			Eventually(func() error {
-				return libnet.PingFromVMConsole(vmi1, cidrToIP(cidrB))
-			}, 15*time.Second, time.Second).Should(Succeed())
-			Eventually(func() error {
-				return libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA))
-			}, 15*time.Second, time.Second).Should(Succeed())
+			assert.XFail("suspected cloud-init issue: https://github.com/kubevirt/kubevirt/issues/4642", func() {
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi1, cidrToIP(cidrB))
+				}, 15*time.Second, time.Second).Should(Succeed())
+				Eventually(func() error {
+					return libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA))
+				}, 15*time.Second, time.Second).Should(Succeed())
```

Comment:
I wonder what is the expected way of using this; what goes in the XFAIL section ? Just the `Expect`ations ? The entire test ? Whatever (weird) call that does not work - e.g. part of the test setup ? 

Having asked that, I think this scoped API makes a lot more sense. Nice improvement.

Reply:
> I wonder what is the expected way of using this; what goes in the XFAIL section ? Just the `Expect`ations ? The entire test ? Whatever (weird) call that does not work - e.g. part of the test setup ?

In practice you can place anything that is executable and it will behave like the ordinary ginkgo flow, but instead of declaring fail, it declares xfail (i.e. skip fail).
As we do not expect non-assert panics and we want to be as specific as possible to what we consider xfail, we better wrap only specific asserts whenever possible.

Reply:
> > I wonder what is the expected way of using this; what goes in the XFAIL section ? Just the `Expect`ations ? The entire test ? Whatever (weird) call that does not work - e.g. part of the test setup ?
> 
> In practice you can place anything that is executable and it will behave like the ordinary ginkgo flow, but instead of declaring fail, it declares xfail (i.e. skip fail).
> As we do not expect non-assert panics and we want to be as specific as possible to what we consider xfail, we better wrap only specific asserts whenever possible.

Right; I understood how it worked, I just wanted to have written somewhere how we're supposed to use this. 

I agree that enclosing the offending assertions is preferable. 

Not sure you want to add that to the commit msg, just a thought. 
 
/lgtm